#162170727 Add like functionality to comments

Johnsonojo opened the pull request:
#### What does this PR do?
It enables users to like or unlike a comment

#### Description of Task to be completed?
 `GET /api/articles/:slug/comments/:commentId/reaction`
* Add a like and unlike contorller
* Add a route for liking and unliking a apecific comment

#### How should this be manually tested?
* clone the repo
* run `npm install`
* change directory to the project folder
* run `npm run dev:start`
* use postman to navigate to any of the endpoints above to like a comment to an article

#### Any background context you want to provide?
* user should be authenticated to like a comment
* user can only like a comment once
* a second attempt to like a comment will result to the user unliking the comment

#### What are the relevant pivotal tracker stories? (If applicable)
[#162170727](https://www.pivotaltracker.com/story/show/162170727)

marcdomain commented on src/routes/articles.js:
```diff
@@ -56,18 +59,15 @@ articlesRouter
 articlesRouter
   .route(`${articlesBaseEndpoint}/:slug/comments/:commentId`)
-  .get(slugChecker, Comment.getOne);
+  // .get(slugChecker, Comment.getOne);
```
Please remove comment

Johnsonojo replied:
Nice observation @marcdomain. Will do that

Johnsonojo replied:
@marcdomain it has been fixed. Thanks

marcdomain reviewed (commented):
LGTM!
Please attend to the comment I left for you on `line 69 of routes/articles.js`

Johnsonojo committed:
[Feature #162170727] Add like functionality to comments

- GET /api/articles/:slug/comments/:commentId/likes
- Add a like and unlike middleware and contorller
- Add route for liking and unliking a comment

[Finishes #162170727]

okoroemeka commented on src/controllers/commentReaction.js:
```diff
@@ -0,0 +1,46 @@
+
+      const commentReaction = await CommentReaction.findOne({ where });
+      if (commentReaction) {
+        await CommentReaction.destroy({ where });
```
Nice job @Johnsonojo and @frostyblok . But I think destroying and recreating a comment reaction is not very effective. I suggest you update the `IsLiked` field instead. I will like to know what you think.

marcdomain replied:
I just thought about @okoroemeka suggestion. I think it's more effective not to delete the row after a user unlike a comment. The implication is that; every new like would create a row which would be assigned a unique id (even if the like and dislike is happening on the same comment, by the same user). Hence, if I liked and disliked a comment 50 times, I'd consume 50 IDs on your model.
Also, please change the `id` dataType to `BIGINT` so that it accommodates more likes, since we'd definitely have more likes than comments and articles in this project.

Johnsonojo replied:
@okoroemeka and @marcdomain, don't you think adding new rows to the table when a user likes and unlikes a comment is not necessary because we're not interested in the number of likes a particular user has given a specific comment. A user can only like a comment once and the next attempt to like will result in an unlike. another thing is having like and unlike rows created every time that action is triggered will just populate the table unnecessarily.

marcdomain replied:
Yeah @Johnsonojo that's the point we're actually driving. However, to make it clearer: A new row should be created when a user likes a comment, then the status of isLiked should change to false (rather than deleting the row) when the user unlikes the comment. If you delete the row, then a new row will always be created for the same user on that same comment, which is not efficient.
So, in essence, updating the row is better than deleting the row.

marcdomain reviewed (changes requested):
Please check the comments on `line 26` of `src/controllers/commentReaction.js`.
Sorry I didn't think it through in this dimension earlier.
